src/lhttpc_lb.erl

```diff
@@ -166,27 +181,26 @@ terminate(_Reason, #state{host=H, port=P, ssl=Ssl, free=Free, clients=Tid}) ->
 %% Potential race condition: if the lb shuts itself down after a while, it
 %% might happen between a read and the use of the pid. A busy load balancer
 %% should not have this problem.
--spec find_lb(Name::{host(),port_number(),boolean()}, {max_connections(), connection_timeout()}) -> pid().
+-spec find_lb(Name::{host(),port_number(),boolean()},
+              {max_connections(), connection_timeout()}) -> pid().
 find_lb(Name = {Host,Port,Ssl}, Args={MaxConn, ConnTimeout}) ->
-    case ets:lookup(?MODULE, Name) of
-        [] ->
-            case supervisor:start_child(lhttpc_sup, [Host,Port,Ssl,MaxConn,ConnTimeout]) of
+    case find_lb(Name) of
+        {error,undefined} ->
+            case supervisor:start_child(lhttpc_sup,
+                                        [Host,Port,Ssl,MaxConn,ConnTimeout]) of
                 {ok, undefined} -> find_lb(Name,Args);
                 {ok, Pid} -> Pid
             end;
-        [{_Name, Pid}] ->
-            case is_process_alive(Pid) of % lb died, stale entry
-                true -> Pid;
-                false ->
-                    ets:delete(?MODULE, Name),
-                    find_lb(Name,Args)
-            end
+        {ok, Pid} ->
+            Pid
```

Comment:
Ah good, sorry. I was just looking at the diff and forgot that bit. Nevermind me here!